packages/lit-html/src/directives/live.ts

```diff
@@ -33,31 +51,25 @@ class LiveDirective extends Directive {
     }
   }
 
-  render(value: unknown) {
+  render(value: unknown, _equal?: EqualFn) {
     return value;
   }
 
-  override update(part: AttributePart, [value]: DirectiveParameters<this>) {
+  override update(
+    part: AttributePart & {
+      readonly type: ValidPartType;
+    },
+    [value, equal]: DirectiveParameters<this>
+  ) {
     if (value === noChange || value === nothing) {
       return value;
     }
-    const element = part.element;
+    const element = part.element as Element;
     const name = part.name;
+    const _equal = equal ?? defaults[part.type];
+
+    if (_equal(value, getValue[part.type](element, name))) return noChange;
```

Comment:
I think we can inline the getValue function to avoid some property lookups:

```suggestion
    const {type, name, element} = part;
    
    const currentValue =
      type === PartType.PROPERTY
        ? element[name as keyof typeof element]
        : type === PartType.ATTRIBUTE
          ? element.getAttribute(name)
          : element.hasAttribute(name);
    if (equal(value, currentValue)) {
      return noChange;
    }
```

Reply:
Updated.